packages/evolution-backend/src/api/survey.user.routes.ts

```diff
@@ -211,29 +212,13 @@ export default (authorizationMiddleware, loggingMiddleware: InterviewLoggingMidd
         }
     );
 
-    router.post('/survey/clientSideException/', async (req: Request, res: Response) => {
+    router.post('/survey/logClientSideMessage/', async (req: Request, res: Response) => {
         // Try/catch to avoid undocumented default behavior in case of an exception, even if we don't expect one
         try {
-            // FIXME: Extract this to a function when we do more than just console.error
             const content = req.body;
-            const interviewId = content?.interviewId || -1;
-            if (content?.exception) {
-                const exceptionString =
-                    typeof content.exception !== 'string' ? String(content.exception) : content.exception;
-                // Log up to 1000 characters of the exception to avoid spamming the logs
-                console.error(
-                    'Client-side exception in interview %d: %s',
-                    interviewId,
-                    exceptionString.substring(0, 1000)
-                );
-            } else {
-                console.error(
-                    'Client-side exception in interview %d: missing exception but content was \'%s\'',
-                    interviewId,
-                    String(content).substring(0, 1000)
-                );
-            }
-            return res.status(200);
+            logClientSide(content);
+            // Need to return a json response status, otherwise the client will wait undefinitely and the survey will block
```

Comment:
The client really need a time out for that. It's easy to lose a request on the internet. You might need to add a TODO or create a new issue.

Reply:
The client is not even waiting for the request, but strangely, if I don't return a json here, after 10 requests, it would just hang, with a bunch of never finished requests in the network tab. I think it's an express thing, we need to "finish" the request by sending data, otherwise the server becomes unresponsive? It could be a text also, but the status is not enough. That was an error in the original code, but we just never reached the number of requests that logging the devLog made ;-)